app/vacancy_sources/every_vacancy_source.rb

```diff
@@ -0,0 +1,113 @@
+class EveryVacancySource
+  FEED_URL = ENV.fetch("VACANCY_SOURCE_EVERY_FEED_URL").freeze
+  SOURCE_NAME = "every".freeze
+
+  class EveryImportError < StandardError; end
+
+  include Enumerable
+
+  def self.source_name
+    SOURCE_NAME
+  end
+
+  def each
+    results.each do |result|
+      v = Vacancy.find_or_initialize_by(
+        external_source: SOURCE_NAME,
+        external_reference: result["reference"],
+      )
+
+      # An external vacancy is by definition always published
+      v.status = :published
+      # Consider publish_on date to be the first time we saw this vacancy come through
+      # (i.e. today, unless it already has a publish on date set)
+      v.publish_on ||= Date.today
+
+      begin
+        v.assign_attributes(attributes_for(result))
+      rescue ArgumentError => e
+        v.errors.add(:base, e)
+      end
```

Comment:
Wouldn't we rather see an exception here so we can fix it?

Reply:
Normally yes, but error handling and reporting is handled higher up here:
That vacancy will be yielded back to the caller which will report on it based on its validity.

https://github.com/DFE-Digital/teaching-vacancies/blob/main/app/jobs/import_from_vacancy_sources_job.rb#L18